### ISSUE TITLE
basic_cancel default method parameters will result in an exception.

### ISSUE THREAD

**Issue by lukebakken** [pika/pika#925]
See [this comment](https://github.com/pika/pika/issues/817#issuecomment-358736838) in issue #817

**Comment by lukebakken** [comment]
This is the default for the following methods as well

```
pika/channel.py:202:    def basic_cancel(self, callback=None, consumer_tag='', nowait=False):
pika/channel.py:542:    def confirm_delivery(self, callback=None, nowait=False):
pika/channel.py:844:    def queue_purge(self, callback=None, queue='', nowait=False):
```

**Comment by lukebakken** [comment]
Re-opening because other methods that default `nowait=False` do not check their callback parameter.

**Comment by lukebakken** [comment]
See #688

**Comment by vitaly-krugl** [comment]
@lukebakken, this should be fixed now because we removed the `nowait` parameter, right?

**Comment by lukebakken** [comment]
Resolved by #941
